### ISSUE TITLE
Spectator mode boots me out of the dimension

### ISSUE THREAD

**Issue by katubug** [CompactMods/CompactMachines#413]
**Compact Machines version**: compactmachines3-1.12.2-3.0.18-b278
**Forge version**: forge-1.12.2-14.23.5.2838-universal
**I am not using Optifine**: Correct.

**Description of the problem including expected versus actual behavior:**
So, I'm a server admin. I've been troubleshooting lag issues using LagGoggles, which has a function that teleports you directly to a specified Tile Entity. Unfortunately, this usually teleports you *into* blocks, because TEs are usually machines connected to pipes. And, usually, these are against the walls of the player's Compact Machines, which leaves me trapped.

Since I can't move, I don't know which coordinates to `/tp` to safely, and I don't want to break any blocks to free myself, as it might disrupt someone's setup that I don't know how to replace correctly. So I've attempted to use `/gamemode spectator` to get unstuck and take a look at the TE in question, but this boots me out of the dimension to my last overworld location.

I'm assuming this is a side effect of this config option?

```
  # Block players from leaving the Compact Machine they should currently be in [default: true]
    B:keepPlayersInside=true
```

But if that's the case, is there a way to allow only ops to bypass this, rather than setting it to false for everyone? I still want to ensure that my players don't accidentally glitch outside their machines and fall to their death.

Thanks for your time!

**Comment by katubug** [comment]
I have confirmed that setting that config to false allows spectator mode. I'm glad there's a workaround, but ideally I'd still like to have different permissions for server ops, if that's possible.

**Comment by robotgryphon** [comment]
Should be able to be resolved with #439, will close this when that gets merged
